src/event/trigger.js

```diff
@@ -136,7 +139,17 @@ jQuery.extend( jQuery.event, {
 
 					// Prevent re-triggering of the same event, since we already bubbled it above
 					jQuery.event.triggered = type;
+
+					if ( event.isPropagationStopped() ) {
+						lastElement.addEventListener( type, stopPropagationCallback );
```

Comment:
@dmethvin One option with the `.stopImmediatePropagation()` would be to register an additional event handler on the second to last element (the great ancestor) if there is one present, that stops the propagation.

But this seems rather fiddly.